common/persistence/src/main/resources/db/migration/V7__createFederationUploadKeyTable.sql

```diff
@@ -0,0 +1,12 @@
+CREATE TABLE federation_upload_key (
+    key_data bytea PRIMARY KEY,
+    rolling_period integer NOT NULL,
+    rolling_start_interval_number integer NOT NULL,
+    submission_timestamp bigint NOT NULL,
+    transmission_risk_level integer NOT NULL,
+    consent_to_federation boolean NOT NULL DEFAULT FALSE,
+    origin_country varchar (2),
```

Comment:
Are we expecting any other value than "DE" here at any point in time?

Reply:
You are right, some fields could be defaulted, including consent_to_federation, which is always true for uploadable keys. It was easier to clone the entire table for the db procedure which replicates the data on submission.  
Also, to point out here is that the implementation of the upload process uses the DiagnosisKey class to map an object to this particular table. This was mainly for ease of development, and because the two data structures (internal diagnosis key and uploadable diagnosis key) do not differ conceptually for this first release. They could diverge in the future though and we  could look into reducing the number of fields to the absolute required.